tests/parser_test.py

```diff
@@ -264,3 +268,155 @@ def test_suffix_precedence() -> None:
+
+KEYWORD_TESTS = [
+    (keyword, t.format(keyword=keyword), r)
+    for (t, r) in [
+        (
+            "for some {keyword} in Variable => {keyword} + 1",
+            librflxlang.GrammarRule.quantified_expression_rule,
+        ),
+        (
+            "[for {keyword} in Variable => {keyword} - 1 when {keyword} > 10]",
+            librflxlang.GrammarRule.comprehension_rule,
+        ),
+        (
+            "{keyword} (Variable + 1)",
+            librflxlang.GrammarRule.call_rule,
+        ),
+        (
+            "{keyword} => Data",
+            librflxlang.GrammarRule.message_aggregate_association_rule,
+        ),
+        (
+            "Data.{keyword}",
+            librflxlang.GrammarRule.extended_suffix_rule,
+        ),
+        (
+            "Data where {keyword} = 100",
+            librflxlang.GrammarRule.extended_suffix_rule,
+        ),
+        (
+            "{keyword} => True",
+            librflxlang.GrammarRule.aspect_rule,
+        ),
+        (
+            "then {keyword} with First => Previous'First if Previous > 100",
+            librflxlang.GrammarRule.then_rule,
+        ),
+        (
+            "{keyword} : Some::Type;",
+            librflxlang.GrammarRule.component_item_rule,
+        ),
+        (
+            "{keyword} => (1, 30..34)",
+            librflxlang.GrammarRule.checksum_association_rule,
+        ),
+        (
+            "{keyword}, {keyword}, Elem_1, Elem_2",
+            librflxlang.GrammarRule.positional_enumeration_rule,
+        ),
+        (
+            "{keyword} => 42",
+            librflxlang.GrammarRule.element_value_association_rule,
+        ),
+        (
+            "type {keyword} is new Other_Type",
+            librflxlang.GrammarRule.type_declaration_rule,
+        ),
+        (
+            "for Some::{keyword} use ({keyword} => Some::{keyword}) if {keyword} > 100",
+            librflxlang.GrammarRule.type_refinement_rule,
+        ),
+        (
+            "type {keyword} is private",
+            librflxlang.GrammarRule.private_type_declaration_rule,
+        ),
+        (
+            "{keyword} : Some::{keyword}",
+            librflxlang.GrammarRule.function_parameter_rule,
+        ),
+        (
+            "with function {keyword} return Some::{keyword}",
+            librflxlang.GrammarRule.formal_function_declaration_rule,
+        ),
+        (
+            "{keyword} : Channel with Readable, Writable",
+            librflxlang.GrammarRule.channel_declaration_rule,
+        ),
+        (
+            "with Initial => {keyword}, Final => {keyword}",
+            librflxlang.GrammarRule.session_aspects_rule,
+        ),
+        (
+            "{keyword} : Some::Type renames Fun",
+            librflxlang.GrammarRule.renaming_declaration_rule,
+        ),
+        (
+            "{keyword} : Some::Type := 42",
+            librflxlang.GrammarRule.variable_declaration_rule,
+        ),
```

Comment:
Type identifiers do not need to be qualified in these cases, thus `Some::{keyword}` and `Some::Type` should be replaceable by `{keyword}`.

Reply:
Changed in 3d4894f